7-gce/tests/test_end_to_end.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import time
+
+from bs4 import BeautifulSoup
+import pytest
+import requests
+
+
+@pytest.mark.e2e
+def test_end_to_end():
+    """Tests designed to be run against live environments.
+
+    Unlike the integration tests in the other packages, these tests are
+    designed to be run against fully-functional live environments.
+
+    To run locally, start both main.py and psq_worker main.books_queue and
+    run this file.
+
+    It can be run against a live environment by setting the E2E_URL
+    environment variables before running the tests:
+
+        E2E_URL=http://your-app-id.appspot.com \
+        nosetests tests/test_end_to_end.py
+    """
+
+    base_url = os.environ.get('E2E_URL', 'http://localhost:8080')
+
+    book_data = {
+        'title': 'a confederacy of dunces',
+    }
+
+    response = requests.post(base_url + '/books/add', data=book_data)
+
+    # There was a 302, so get the book's URL from the redirect.
+    book_url = response.request.url
+    book_id = book_url.rsplit('/', 1).pop()
+
+    # Wait 30 seconds for pubsub messages to be processed.
+    time.sleep(30)
+
```

Comment:
60 seconds is the ack deadline, so this may actually be too short. I think it's a safe value for now and we can adjust. These tests will only run on jenkins anyway.